packages/core/src/extension/resizable-iframe-creator.ts

```diff
@@ -41,9 +47,13 @@ export default class ResizableIframeCreator {
         // Instead, listen to the `load` inside the iframe and let the parent frame know when it happens.
         return new Promise((resolve, reject) => {
             const timeout = window.setTimeout(() => {
+                this._extensionMessenger.post(this._extensionId, {
```

Comment:
Can we just create a new instance of `IframeEventPoster` and use it to post the `ExtensionInternalEventType.ExtensionFailed` event and `ExtensionInternalEventType.ExtensionReady` event? That way we don't have to retrofit `ExtensionMessenger` to post both `ExtensionInternalEvent` and `ExtensionEvent`. That change could be problematic because callers of `ExtensionMessenger#post` method expect to post one or the other event, not both.

Reply:
`ExtensionMessenger` won't be used here. A one-time use `IframeEventPoster` will be created in the callback function.